new/rfcXXX-lifecycle-and-release-policies.md

```diff
@@ -0,0 +1,214 @@
+
+##### Semantic Versioning Exceptions
+
+In the case where Chef OSS that would otherwise fall under the scope of the document does not follow Semantic Versioning, the exceptions will be noted below with a description of the versioning scheme they follow.
+
+| Software | Reasoning |
+|---|---|
+| [delivery CLI](https://github.com/chef/delivery-cli) | The delivery CLI is a Rust utility bundled with the ChefDK. When it is built, it is always built off of master. It's versioning scheme is based on the Git SHA that was compiled. |
+
+#### Artifact Build
+
+Chef OSS artifacts available on [https://downloads.chef.io](https://downloads.chef.io) are built using Chef Software's internal CI/CD build system. Depending on the software project, additional artifacts such as Ruby Gems or Docker Images may also be compiled by ancillary build systems and published to their appropriate locations. Instructions for how to install those additional artifacts will be made available on the projects README.
+
+The commit used to generate an artifact will be tagged with the corresponding `MAJOR.MINOR.PATCH` version. The specific formatting of the tag can vary as long as the three version segments are present (eg. "v1.2.3", "1_2_3", or "1.2.3" are all valid) This will allow users, developers, and support personnel to easily navigate the source code associated with a particular build or release.
+
+Some Chef OSS artifacts (eg. ancillary gems) are built and published manually by their maintainers.
+
+Some Chef OSS will be built following a Continuous Integration / Continuos Delivery (CI/CD) model. This means that every merge to master may result in a build and every successful build that passes functional testing is a candidate for a Stable Release and published for consumption by users. This is especially common for projects that are under Cadence Releases.
+
+Some Chef OSS projects will have very large sets of dependencies that are allowed to float (eg. ~> version constraint). Doing so allows other projects to consume them without having to match dependencies in lock-step, helping to prevent ecosystem failure when project upgrades before another. Because these dependencies are resolved at build-time, two builds of the same Chef OSS commit may result in slightly different packages. It is for this reason that we encourage users, when possible, to consume pre-compiled binaries through official channels (where this effect is minimized) rather than compiling them themselves.
+
+#### Build Channels
+
+Chef Software maintains three artifact channels: unstable, current, and stable.
+
+  * unstable: Builds that have completed successfully and are awaiting additional automated testing.
+  * current: Builds that have passed all automated testing.
```

Comment:
This might be slightly off since the promotion process means it is all builds that have passed automation and are not a stable release. It bites people from time to time, but I think the plan is to fix that? (i.e. make promotion a copy rather than a move, or did that already happen?)